Purchases/FoundationExtensions/Result+Extensions.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Result+Extensions.swift
+//
+//  Created by Nacho Soto on 12/1/21.
+
+extension Result {
+    var value: Success? {
+        switch self {
+        case let .success(value): return value
+        case .failure: return nil
+        }
+    }
+
+    var error: Failure? {
+        switch self {
+        case .success: return nil
+        case let .failure(error): return error
+        }
+    }
+}
```

Comment:
> The upside of using Result + this is that we're guaranteed to have either value or error, never both, never neither.

Right, and since these APIs are only internal, therefore Swift only, I think it was the best solution.

> The downside being that it makes it so much easier to not think about error handling,
> when you have to do a couple of switch cases to get the value, you're absolutely forced to think about the error case, which I like, even if you then decide to do nothing with the error.

That's absolutely right. I tried to look at every place where we were ignoring errors and using `Set([])`, but I realized that it would be a lot of work right now to handle each situation.

I think the extensions can be useful in some places, but maybe I should change most of the places where I used them in this PR and have the explicit `case let .failure`? Or at least leave a comment in each?